### PR TITLE
added bubbling to hooked wallet for idmgmt layers

### PR DESCRIPTION
**Pull request by zmitton**
@kumavis after we talked on slack I took a look at the code and wanted to see if there was any easy way to have the wallet work in layers how I expected it. I definitely don't know fully how this thing works, but seemed to actually get somewhere with it. Is this the right idea for implementing layers?

### CODE REVIEW

File: subproviders/hooked-wallet.js [hunk 1]
```diff
@@ -60,33 +69,42 @@ HookedWalletSubprovider.prototype.handleRequest = function(payload, next, end){
...
+        if (err) return cb(err)
+        if(manages){
+          var txParams = payload.params[0]
+          async.waterfall([
+            self.validateTransaction.bind(self, txParams),
+            self.approveTransaction.bind(self, txParams),
+            self.checkApproval.bind(self),
+            self.finalizeAndSubmitTx.bind(self, txParams)
+          ], end)
+        }else{ next() }
```
Comment (flyswatter) [inline subproviders/hooked-wallet.js]: if the account is not managed, would it make more sense to end() with an error describing the reason?

That error could even be sent from the mamagesAccount method, maybe it should be onlyIfManages or something in that case, then you don't need the conditional,  just the error handling.

**Comment by flyswatter** [comment]
Also I think by "bubbling/layers" you were probably meaning to create a new, account-validating subprovider?

The consumer would have to manage accounts in that case, so that would be a pain. HookedWallet is the easier place to do this, but this is also newly opinionated, and so a breaking change.

A new subprovider would let provider-engine consumers choose whether to opt in to the new validation, and it could be default in ZeroClient, but more to config.

**Comment by zmitton** [comment]
@flyswatter actually, Aaron said it needs a constructor option that allows fallthrough. So would that fix the problem?

**Comment by flyswatter** [comment]
Yeah, if you had a constructor argument that defaulted to the old behavior, that would fix my concern.

But my other point about maybe throwing an error still stands.

**Comment by zmitton** [comment]
I don't think i follow. wouldn't an error, stop the propagation into the next() subprovider?

**Comment by flyswatter** [comment]
Yes, you'd call `end(new Error('User cannot sign with an account they do not manage.'))`.

Stopping propagation would be the point, because passing an unsigned tx is inherently failing the method.

**Comment by flyswatter** [comment]
Closing for now, seems to have lost momentum, feel free to re-submit if this ever gets worked out.

**Pull request closed without merging**